package-res/resources/web/pentaho/type/_type.js

```diff
@@ -0,0 +1,1362 @@
+       *
+       *   var context = new Context({application: "data-explorer-101"});
+       *   var ProductList = context.get([{
+       *           props: [
+       *             "id",
+       *             "name",
+       *             {name: "price", type: "number"}
+       *           ]
+       *         }]);
+       *
+       *   // Provide the default type, in case the instance spec doesn't provide one.
+       *   var productList = ProductList.type.create(
+       *          [
+       *            {id: "mpma", name: "Principia Mathematica", price: 1200},
+       *            {id: "flot", name: "The Laws of Thought",   price:  500}
+       *         ]);
+       *
+       *   // ...
+       *
+       * });
+       *
+       * @param {pentaho.type.spec.UInstance} [instSpec] An instance specification.
+       *
+       * @return {!pentaho.type.Instance} The created instance.
+       *
+       * @throws {pentaho.lang.OperationInvalidError} When `instSpec` contains an inline type reference
+       * that refers to a type that is not a subtype of this one.
+       *
+       * @throws {pentaho.lang.OperationInvalidError} When the determined type for the specified `instSpec`
+       * is an [abstract]{@link pentaho.type.Value.Type#isAbstract} type.
+       */
+      create: function(instSpec) {
+        var Instance, typeSpec;
+
+        // If it is a plain Object, does it have the inline type property, "_"?
+        if(instSpec && typeof instSpec === "object" && (typeSpec = instSpec._) && instSpec.constructor === Object) {
+
+          Instance = context.get(typeSpec);
+
+          if(this._assertSubtype(Instance.type).isAbstract)
+            Instance.type._throwAbstractType();
+
+        } else {
+          if(this.isAbstract) this._throwAbstractType();
+
+          // Does this type have an own constructor?
+          var baseInst = this.instance;
+
+          Instance = baseInst.constructor;
+
+          if(Instance.prototype !== baseInst) {
+            // Type was created through extendProto.
+            var inst = Object.create(baseInst);
+            return Instance.apply(inst, arguments) || inst;
+          }
+        }
+
+        return O.make(Instance, arguments);
+      },
+
+      /**
+       * Asserts that a given type is a subtype of this type.
+       *
+       * @param {!pentaho.type.Type} subtype The subtype to assert.
+       *
+       * @return {!pentaho.type.Type} The subtype `subtype`.
+       *
+       * @throws {pentaho.lang.OperationInvalidError} When `subtype` is not a subtype of this.
+       *
+       * @private
+       */
+      _assertSubtype: function(subtype) {
+        if(!subtype.isSubtypeOf(this)) {
+          throw error.operInvalid(
+              bundle.format(bundle.structured.errors.instance.notOfExpectedBaseType, [this._getErrorLabel()]));
+        }
+
+        return subtype;
+      },
+
+      /**
+       * Throws an error complaining the type is abstract an cannot create instances.
```

Comment:
Suugest:

Throws an error complaining the type is abstract and cannot create instances.